crypto/eth2_wallet/src/wallet.rs

```diff
@@ -285,3 +285,16 @@ pub fn recover_validator_secret(
 
     Ok((destination.secret().to_vec().into(), path))
 }
+
+pub fn recover_validator_secret_from_mnemonic(
```

Comment:
Suggestion for comment:

```
/// Returns `(secret, path)` for the `key_type` for the validator at `index`.
///
/// This function should only be used for key recovery since it can easily lead to key duplication.
```